tcms/testplans/tests/test_new_plan.py

```diff
@@ -1,73 +1,115 @@
-
-        user_should_have_perm(self.user, perm=self.add_testplan_permission)
-
-        self.user.is_superuser = False
-        self.user.save()
-
-        self.request = {
-            'author': self.user.pk,
-            'product': self.product.pk,
-            'product_version': self.product_version.pk,
-            'type': self.plan_type.pk,
-            'name': self.plan_name,
-
-            'email_settings-0-auto_to_plan_author': 'on',
-            'email_settings-0-auto_to_case_owner': 'on',
-            'email_settings-0-auto_to_case_default_tester': 'on',
-            'email_settings-0-notify_on_case_update': 'on',
-            'email_settings-0-notify_on_plan_update': 'on',
-
-            'email_settings-0-plan': '',
-            'email_settings-0-id': self.test_plan.emailing.pk,
-            'email_settings-TOTAL_FORMS': '1',
-            'email_settings-INITIAL_FORMS': '1',
-            'email_settings-MIN_NUM_FORMS': '0',
-            'email_settings-MAX_NUM_FORMS': '1',
+from tcms.tests import (
+    remove_perm_from_user,
+    user_should_have_perm,
+    LoggedInTestCase,
+    PermissionsTestCase,
+)
+from tcms.tests.factories import (
+    TestPlanFactory,
+    ProductFactory,
+    VersionFactory,
+    PlanTypeFactory,
+)
+
+
+class TestNewTestPlanView(LoggedInTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.product = ProductFactory()
+        cls.product_version = VersionFactory(product=cls.product)
+        cls.test_plan_type = PlanTypeFactory()
+        cls.test_plan_1 = TestPlanFactory()
+
+        user_should_have_perm(cls.tester, perm="testplans.add_testplan")
+        user_should_have_perm(cls.tester, perm="testplans.view_testplan")
+
+        cls.url = reverse("plans-new")
+
+        cls.request = {
+            "author": cls.tester.pk,
+            "product": cls.product.pk,
+            "product_version": cls.product_version.pk,
+            "type": cls.test_plan_type.pk,
+            "name": cls.test_plan_1.name,
+            "email_settings-0-auto_to_plan_author": "on",
+            "email_settings-0-auto_to_case_owner": "on",
+            "email_settings-0-auto_to_case_default_tester": "on",
+            "email_settings-0-notify_on_case_update": "on",
+            "email_settings-0-notify_on_plan_update": "on",
+            "email_settings-0-id": cls.test_plan_1.emailing.pk,
+            "email_settings-TOTAL_FORMS": "1",
+            "email_settings-INITIAL_FORMS": "1",
+            "email_settings-MIN_NUM_FORMS": "0",
+            "email_settings-MAX_NUM_FORMS": "1",
+            "is_active": True,
         }
 
     def test_plan_new_get(self):
-        response = self.client.get(self.location)
+        response = self.client.get(self.url)
 
         self.assertContains(
             response,
             '<input class="bootstrap-switch" name="is_active" type="checkbox" checked',
-            html=False)
+            html=False,
+        )
```

Comment:
Why do you need all of these unrelated changes ?

Reply:
I don't quite understand what you mean by unrelated, can you please elaborate? (I'm sorry I'm a little slow to understand things)

Reply:
many of the changes you have made are updating formatting only and are not related to the issue in question nor are adding new automated tests. Their only purpose is to change formatting of existing code.

So as the first step I am asking why do we need them added in this pull request and what necessitates these changes ?

Reply:
Yeah, about the formatting changes, it's my bad, I accidentally ran black on my changes, which did this.